lib/ghomfp.gi

```diff
@@ -197,30 +197,38 @@ end);
 InstallMethod(CosetTableFpHom,"for fp homomorphisms",true,
   [ IsFromFpGroupGeneralMappingByImages and IsGroupGeneralMappingByImages],0,
 function(hom)
-local u,aug,hgu,mapi;
+local u,aug,hgu,mapi,w;
   # source group with suitable generators
   u:=Source(hom);
   aug:=false;
   mapi:=MappingGeneratorsImages(hom);
   hgu:=List(mapi[1],UnderlyingElement);
-  # try to re-use an existing augmented coset table:
-  aug:=AugmentedCosetTableInWholeGroup(u,mapi[1]);
-  if not IsSubset(hgu,aug.primaryGeneratorWords) then
-    # we don't know what to do with the extra primary words, so enforce MTC
-    # version
-    aug:=AugmentedCosetTableMtcInWholeGroup(
-           SubgroupNC(FamilyObj(u)!.wholeGroup,mapi[1]));
-  fi;
-  # as we add homomorphism specific entries, lets be safe and copy.
-  aug:=CopiedAugmentedCosetTable(aug);
+
+  # construct augmented coset table
+  w:=FamilyObj(mapi[1])!.wholeGroup;
+  aug:=NEWTC_CosetEnumerator(FreeGeneratorsOfFpGroup(w),
+	RelatorsOfFpGroup(w),
+	hgu,
+	true);
 
   aug.homgens:=mapi[1];
   aug.homgenims:=mapi[2];
 
   # assign the primary generator images
-  aug.primaryImages:=List(aug.primaryGeneratorWords,
+  aug.primaryImages:=List(aug.subgens,
                           i->aug.homgenims[Position(hgu,i)]);
 
+  # # try to re-use an existing augmented coset table:
+  # aug:=AugmentedCosetTableInWholeGroup(u,mapi[1]);
+  # if not IsSubset(hgu,aug.primaryGeneratorWords) then
+  #   # we don't know what to do with the extra primary words, so enforce MTC
+  #   # version
+  #   aug:=AugmentedCosetTableMtcInWholeGroup(
+  #          SubgroupNC(FamilyObj(u)!.wholeGroup,mapi[1]));
+  # fi;
+  # as we add homomorphism specific entries, lets be safe and copy.
+  # aug:=CopiedAugmentedCosetTable(aug);
+
   TrySecondaryImages(aug);
```

Comment:
What is this newly added but commented out code about? Is this a deadend? Then it should be removed. Is it a place where a future extension is planned? Then it would be good to state that.

Otherwise, we'll just drag this around forever, and in years somebody will wonder what it's about,but unable to find out details...